src/CPM/MDKoutage.py

```diff
@@ -0,0 +1,82 @@
+import pyomo.environ as pyo
+import numpy as np
+import random
+
+
+class mdkChoiceModel:
+    """
+        This is the base class for the multi-dimensional knapsack problem adapted to the outage
+        scheduling problem
+    """
+    def __init__(self, candidates, resources, valueType):
+        """
+        Constructor
+        @ In, candidates, list, list of candidate activities
```

Comment:
From the following, the candidates is a dict instead

Reply:
fixed